MakieCore/src/conversion.jl

```diff
@@ -30,3 +30,103 @@ conversion_trait(::Type{<: Heatmap}) = DiscreteSurface()
 
 struct VolumeLike <: ConversionTrait end
 conversion_trait(::Type{<: Volume}) = VolumeLike()
+
+
+"""
+    convert_arguments_typed(::Type{<: AbstractPlot}, args...)::NamedTuple
+
+Converts the arguments to their correct type for the Plot type.
+Throws apropriate errors if it can't convert!
+"""
+function convert_arguments_typed end
+
+
+function convert_arguments_typed(P::Type{<: AbstractPlot}, @nospecialize(args...))
+    convert_arguments_typed(conversion_trait(P), args...)
+end
+
+function convert_arguments_typed(::CT, @nospecialize(args...)) where {CT <: ConversionTrait}
```

Comment:
why this redirect? the functions should either take CT or CT(), not both

Reply:
The conversion pipeline has been using `::Type{<:AbstractPlot}` and `::ConversionTrait` for dispatch, but in the macro I can't differentiate between those, so I overload `::Type{<:T}` for both.
I guess I could also use `typeof(conversion_trait(P))`, instead of redirecting.